components/service/glean/src/main/java/mozilla/components/service/glean/CounterMetricType.kt

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean
+
+import mozilla.components.service.glean.storages.CountersStorageEngine
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * This implements the developer facing API for recording string metrics.
+ *
+ * Instances of this class type are automatically generated by the parsers at build time,
+ * allowing developers to record values that were previously registered in the metrics.yaml file.
+ *
+ * The string API only exposes the [set] method, which takes care of validating the input
+ * data and making sure that limits are enforced.
+ */
+data class CounterMetricType(
+    override val disabled: Boolean,
+    override val category: String,
+    override val lifetime: Lifetime,
+    override val name: String,
+    override val sendInPings: List<String>
+) : CommonMetricData {
+
+    override val defaultStorageDestinations: List<String> = listOf("metrics")
+
+    private val logger = Logger("glean/CounterMetricType")
+
+    private var count = 0
```

Comment:
Since it's possible that the value is stored in multiple locations due to how the 'sendInPings' value is a list type, can I safely assume all those stored values are the same and just grab the first one?

Reply:
That's a good question.  They very well might be different.  It might be easiest to actually handle the addition in the storage engine itself, to be honest.  I don't know if that means the `GenericScalarStorageEngine` base class will become less helpful or need some refactoring...

Reply:
Using custom combiner from #1476 implementation solves this issue